cmd/orb.go

```diff
@@ -740,3 +758,144 @@ https://circleci.com/orbs/registry/orb/%s
 
 	return nil
 }
+
+type OrbSchema struct {
+	Version     string                   `yaml:"version,omitempty"`
+	Description string                   `yaml:"description,omitempty"`
+	Display     interface{}              `yaml:"display,omitempty"`
+	Orbs        interface{}              `yaml:"orbs,omitempty"`
+	Commands    interface{}              `yaml:"commands,omitempty"`
+	Executors   interface{}              `yaml:"executors,omitempty"`
+	Jobs        interface{}              `yaml:"jobs,omitempty"`
+	Examples    map[string]ExampleSchema `yaml:"examples,omitempty"`
+}
+
+type ExampleUsageSchema struct {
+	Version   string      `yaml:"version,omitempty"`
+	Orbs      interface{} `yaml:"orbs,omitempty"`
+	Jobs      interface{} `yaml:"jobs,omitempty"`
+	Workflows interface{} `yaml:"workflows"`
+}
+
+type ExampleSchema struct {
+	Description string             `yaml:"description,omitempty"`
+	Usage       ExampleUsageSchema `yaml:"usage,omitempty"`
+	Result      ExampleUsageSchema `yaml:"result,omitempty"`
+}
+
+func packOrb(opts orbOptions) error {
+	// Travel our Orb and build a tree from the YAML files.
+	// Non-YAML files will be ignored here.
+	_, err := os.Stat(filepath.Join(opts.args[0], "@orb.yml"))
+	if err != nil {
+		return errors.Wrap(err, "@orb.yml file not found, are you sure this is the Orb root?")
+	}
+
+	tree, err := filetree.NewTree(opts.args[0], "executors", "jobs", "commands", "examples")
+	if err != nil {
+		return errors.Wrap(err, "An error occurred trying to build the tree")
+	}
+
+	y, err := yaml.Marshal(&tree)
+	if err != nil {
+		return errors.Wrap(err, "Failed trying to marshal the tree to YAML")
+	}
+
+	// Create generic YAML node.
+	var node yaml.Node
+	err = yaml.Unmarshal(y, &node)
+	if err != nil {
+		return errors.Wrap(err, "Failed unmarshal YAML tree")
+	}
+
+	err = travelOrbTree(&node, opts.args[0])
+	if err != nil {
+		return errors.Wrap(err, "Failed trying to travel Orb YAML")
+	}
+
+	final, err := yaml.Marshal(&node)
+	if err != nil {
+		return errors.Wrap(err, "Failed trying to marshal Orb YAML")
+	}
+
+	// Unmarshal again into a struct so that it can be
+	// marhsalled again but with custom-ordered keys
+	var orb OrbSchema
+	err = yaml.Unmarshal(final, &orb)
+	if err != nil {
+		return errors.Wrap(err, "Failed unmarshal YAML tree")
+	}
+
+	finalOrdered, err := yaml.Marshal(&orb)
+	if err != nil {
+		return errors.Wrap(err, "Failed trying to marshal Orb YAML")
+	}
+
+	fmt.Println(string(finalOrdered))
+
+	return nil
+}
+
+// Travel down a YAML node, replacing values as we go.
+func travelOrbTree(node *yaml.Node, orbRoot string) error {
```

Comment:
nit: I'd prefer a name about what it's doing when it travels instead - `inlineIncludes` or something like that.